File: coding-challenges/sources/leetcode/1. Two Sum/1-TwoSum.py
#!/usr/bin/env python3

from typing import List, Tuple

def two_sum(nums: List[int], target: int) -> Tuple[int, int]:
    dic = dict()
    for index, num in enumerate(nums):
        complement = target - num
        if complement in dic:
            return [dic[complement], index]
        dic[num] = index


print(two_sum([2,7,11,15], 9))

File: coding-challenges/sources/leetcode/1. Two Sum/1-TwoSum.swift
/**
 # [Two Sum](https://leetcode.com/problems/two-sum/)
 */

fileprivate func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var dic = [Int: Int]()
    return nums
        .enumerated()
        .reduce(into: [Int]()) { acc, seq in
            let num = seq.element
            let index = seq.offset
            let difference = target - num
            if let diff = dic[difference] {
                acc = [diff, index]
            }
            dic[num] = index
    }
}

import XCTest

final class TwoSum: XCTestCase {
    func test() {
        print(twoSum([2,7,11,15], 9))
    }
}

